R2.6: Update Compatibility chart, Release Notes and Changelist

inc1t3Ful opened the pull request.

arvindsree commented:
Thank you for the PR

arvindsree commented on aviatrix/resource_gateway.go:
```diff
@@ -185,6 +189,82 @@ func resourceAviatrixGateway() *schema.Resource {
+		GwName:      gateway_name,
+		AccountName: account_name,
+	}
+	gw, err := client.GetGateway(gateway)
```
Better to just just call resourceAviatrixGatewayRead here, as it does the same operation (GetGateway)

Most of the arguments set below are input arguments and are not returned by the GetGateway operation

jalvarezferr replied:
resourceAviatrixGatewayRead does not return all the required attributes. If it did, a custom import function would not be required and the schema.ImportStatePassthrough could be used, because in the state refresh phase performed by Terraform, all the fields would already be fetched.
Hence, al alternative implementation would be modifying resourceAviatrixGatewayRead to get and return all values not present in the received schema.ResourceData, but I chose a separate import to be less intrusive in the existing code.

arvindsree replied:
The problem is that the REST API does not return all attributes. So, setting those parameters are redundant, as of now. 

It would be ideal to use schema.ImportStatePassthrough, if possible

jalvarezferr replied:
I agree, about the use of  schema.ImportStatePassthrough, but for that we need to add to resourceAviatrixGatewayRead all other attributes. It is true that there are many attributes that GetGateway does not return, but that's why I surrounded each attribute with an if to check if it has a value, so if new attributes are returned, the implementation does not need to change.
Would you agree if I add to resourceAviatrixGatewayRead the code to d.Set() all attributes that are returned by the API, and use schema.ImportStatePassthrough as the schema importer?

arvindsree replied:
Sounds good

jalvarezferr replied:
Check latest commits

arvindsree commented on aviatrix/resource_gateway.go:
```diff
@@ -303,12 +328,70 @@ func resourceAviatrixGatewayRead(d *schema.ResourceData, meta interface{}) error
+		    }
+		    d.Set("elb_name", elb_name)
+		} else {
+		    d.Set("enable_elb","no")
```
You can set elb_name to empty here

jalvarezferr replied:
Is it necessary? enve right? If it is set to "" it will appear in the state as elb_name = "", which is awkward in a Terraform state. If it is not set it will not be present in the state, which for me is more aligned with it not being set in the resource.
Originally elb_name was a conditional attribute, as it could be optionally declared as part of the resource definition. Now it is computed, although it is still something that can be declared in the resource creation. I think the original conditional type was the correct one. In that sense, too, at import it would be better to not having it declared in the state if it has not a value.

arvindsree replied:
Ok.. only thing is if someone sets it manually to some other value when elb is disabled, then a "refresh" will not correct it

inc1t3Ful committed:
Change reference back to new release doc

inc1t3Ful committed:
Merge branch 'master' of https://github.com/inc1t3Ful/terraform-provider-aviatrix

inc1t3Ful committed:
(11113) Update doc for versioning references

++ added references to provider version and controller version
** fixed formatting on release compatibility chart
** fixed code block formatting for upgrade guide

inc1t3Ful committed:
Update release compatibility chart with example

inc1t3Ful committed:
Merge branch 'master' of https://github.com/inc1t3Ful/terraform-provider-aviatrix

inc1t3Ful committed:
Merge remote-tracking branch 'upstream/master'

inc1t3Ful committed:
R2.6: Update Compatibility chart, Release Notes and Changelist

arvindsree commented:
why is the refresh section for fireall instance violet?

Do you we support modifying firenet_gw_name in instance associtation in firenet?

inc1t3Ful committed:
R2.6: Added single_az_ha support reference to docs
